cocos/core/math/vec3.ts

```diff
@@ -801,6 +801,32 @@ export class Vec3 extends ValueType {
         return out;
     }
 
+    /**
+     * @zh 生成指定向量的一个正交单位向量。如果指定的向量 **精确地** 是零向量，则返回 **精确的** 零向量。
+     * @en Generates an unit vector orthogonal to specified vector.
+     * If the specified vector is **strictly** zero vector, the result is **strict** zero vector.
+     * @param out @zh 生成的向量。@en The generated vector.
+     * @param n @zh 输入向量。该向量 **不必** 是标准化的。 @en The input vector. **Need not** to be normalized.
+     * @returns `out`
+     */
+    public static generateOrthogonal<Out extends IVec3Like> (out: Out, n: Readonly<IVec3Like>): Out {
```

Comment:
It is better not use IVec3Like.

Reply:
Just use Vec3

Reply:
Why? I just following the existing style

Reply:
Because IVec3Like is duck type. And the original implementation also use Vec3.

Reply:
> Why? I just following the existing style

After discussion, i think it is reasonable. We should modify all the functions at the same time.